bin/restore-standalone.sh

```diff
@@ -0,0 +1,235 @@
+  echo "--bootstrap-server is missing"
+  echo "$HELP"
+  exit 1
+fi
+
+if [ -z "$SOURCE_DIR" ]; then
+  echo "--source-dir is missing"
+  echo "$HELP"
+  exit 1
+fi
+
+if [ ! -d "$SOURCE_DIR" ]; then
+  echo "Directory $SOURCE_DIR does not exist."
+  exit 1
+fi
+
+if [ -z "$TOPICS" ]; then
+  echo "--topics is missing"
+  echo "$HELP"
+  exit 1
+fi
+
+if [ -z "$OFFSET_FILE" ]; then
+  OFFSET_FILE="$SOURCE_DIR/restore.offsets"
+fi
+
+if ! touch "$OFFSET_FILE"; then
+  echo "cannot touch $OFFSET_FILE. Please make sure it is writable"
+  exit 1
+fi
+
+if [ -n "$COMMAND_CONFIG" ] && [ ! -f "$COMMAND_CONFIG" ]; then
+  echo "no such file $COMMAND_CONFIG"
+  exit 1
+fi
+
+if [ ! -f "$PLUGIN_PATH/kafka-backup.jar" ]; then
+  echo "Cannot find the kafka-backup.jar in $PLUGIN_PATH. Please set --backup-jar accordingly"
+  exit 1
+fi
+
+if [ -n "$(command -v connect-standalone.sh)" ]; then
+  CONNECT_BIN="connect-standalone.sh"
+fi
+
+if [ -n "$(command -v connect-standalone)" ]; then
+  CONNECT_BIN="connect-standalone"
+fi
+
+if [ -z "$CONNECT_BIN" ]; then
+  echo "Cannot find connect-standalone or connect-standalone.sh in PATH. please add it"
+  exit 1
+fi
+
+##################################### Create configs
+
+# Standalone Worker Config
+
+WORKER_CONFIG="$WORKING_DIR/standalone.properties"
+
+cat <<EOF >"$WORKER_CONFIG"
+bootstrap.servers=$BOOTSTRAP_SERVER
+key.converter=org.apache.kafka.connect.json.JsonConverter
+value.converter=org.apache.kafka.connect.json.JsonConverter
+key.converter.schemas.enable=false
+value.converter.schemas.enable=false
+offset.storage.file.filename=$WORKING_DIR/kafka-backup.offsets
+offset.flush.interval.ms=10000
+plugin.path=$PLUGIN_PATH
+EOF
+
+if [ -n "$COMMAND_CONFIG" ]; then
+  cat "$COMMAND_CONFIG" >>"$WORKER_CONFIG"
+fi
+
+# Connector config
+
+CONNECTOR_CONFIG="$WORKING_DIR/connector.properties"
+
+cat <<EOF >"$CONNECTOR_CONFIG"
+name=backup-source
+connector.class=de.azapps.kafkabackup.source.BackupSourceConnector
+tasks.max=1
+topics=$TOPICS
+key.converter=de.azapps.kafkabackup.common.AlreadyBytesConverter
+value.converter=de.azapps.kafkabackup.common.AlreadyBytesConverter
+source.dir=$SOURCE_DIR
+batch.size=$BATCH_SIZE
+cluster.bootstrap.servers=$BOOTSTRAP_SERVER
+cluster.key.deserializer=org.apache.kafka.common.serialization.ByteArrayDeserializer
+cluster.value.deserializer=org.apache.kafka.common.serialization.ByteArrayDeserializer
+EOF
+
+if [ -n "$COMMAND_CONFIG" ]; then
+  sed 's/^/cluster./' "$COMMAND_CONFIG" >>"$CONNECTOR_CONFIG"
+fi
+
+if [ "$DEBUG" == "y" ]; then
+  echo "$WORKER_CONFIG:"
+  sed 's/^/> /' "$WORKER_CONFIG"
+  echo
+  echo "$CONNECTOR_CONFIG:"
+  sed 's/^/> /' "$CONNECTOR_CONFIG"
+  echo
+
+fi
+
+##################################### Start Connect Standalone
+"$CONNECT_BIN" "$WORKER_CONFIG" "$CONNECTOR_CONFIG" >>"$WORKING_DIR/log" &
+PID=$!
+tail -F "$WORKING_DIR/log" &
+PID2=$!
+sleep 5
+grep -q "All records read. Restore was successful" <(tail -F "$WORKING_DIR/log" 2>/dev/null)
```

Comment:
Not sure I got this part right.. I see you're trying to find the line in the log and terminate after. Cannot get why 2 `tail -F` on same log file though... 
I'd write it in this way:
```
tail -F "$WORKING_DIR/log" | while read -r line; do
  echo "$line" | grep -q "All records read. Restore was successful" && _term
done
```

Here `tail -F | while read` would block until right log message found.

Reply:
The tail in line 230 is just printing the logs out to `stdout` such that the user can see it. The second tail is piping it to `grep`. I have tried to use tee in line 229 but then the `kill $PID` would not kill Kafka connect.

In general I do not like to use `while read` in bash. It feels awkward to me.

I have added a bunch more comments to clarify what happens. Maybe you have a better idea how to do so ;)